src/psyclone/psyGen.py

```diff
@@ -5840,55 +5882,200 @@ def _number_handler(self, node, parent):
-                          into the kernel.
-    :param bool is_output: Whether the symbol represents data that is passed \
-                           outside the kernel upon exit.
 
     :raises NotImplementedError: Provided parameters are not supported yet.
     :raises TypeError: Provided parameters have invalid error type.
     :raises ValueError: Provided parameters contain invalid values.
 
     '''
-    # Tuple with the valid values for the access attribute.
-    valid_scope_types = ('local', 'global_argument')
-    # Tuple with the valid datatypes.
-    valid_data_types = ('real', 'integer', 'character', 'boolean')
-    # Mapping from supported data types for constant values to
-    # internal Python types
+    ## Tuple with the valid datatypes.
+    valid_data_types = ('real',  # Floating point
+                        'integer',
+                        'character',
+                        'boolean',
+                        'deferred')  # Type of this symbol not yet determined
+    ## Mapping from supported data types for constant values to
+    #  internal Python types
     mapping = {'integer': int, 'character': str, 'boolean': bool}
 
-    def __init__(self, name, datatype, shape=[], scope='local',
-                 constant_value=None, is_input=False, is_output=False):
+    class Access(Enum):
+        '''
+        Enumeration for the different types of access that a Symbol is
+        permitted to have.
+
+        '''
+        ## The symbol is only ever read within the current scoping block.
+        READ = 1
+        ## The first access of the symbol in the scoping block is a write and
+        # therefore any value that it may have had upon entry is discarded.
+        WRITE = 2
+        ## The first access of the symbol in the scoping block is a read but
+        # it is subsequently written to.
+        READWRITE = 3
+        ## The way in which the symbol is accessed in the scoping block is
+        # unknown
+        UNKNOWN = 4
+
+    class Argument(SymbolInterface):
+        '''
+        Captures the interface to a symbol that is accessed as a routine
+        argument.
+
+        :param access: how the symbol is accessed within the local scope.
+        :type access: :py:class:`psyclone.psyGen.Symbol.Access`
+        '''
+        def __init__(self, access=None):
+            super(Symbol.Argument, self).__init__(access=access)
+            self._pass_by_value = False
+
+        @property
+        def is_argument(self):
+            '''
+            :returns: Whether or not this symbol is a routine argument.
+            :rtype: bool
+            '''
+            return True
+
+        def __str__(self):
+            return "Argument(pass-by-value={0})".format(self._pass_by_value)
+
+    class FortranGlobal(SymbolInterface):
+        '''
+        Describes the interface to a Fortran Symbol representing data that
+        exists outside of the local scope but is not supplied as a routine
+        argument.
```

Comment:
Instead of "but is not supplied as a routine argument." why not use "is supplied as a Fortran global variable."
In the future we may want to add more SymbolInterface classes.

Reply:
Done.